### ISSUE TITLE
Porting Webform to ASP.NET Core and run it across operating systems, vote if you think it will help you...

### ISSUE THREAD

**Issue by birojnayak** [CoreWebForms/CoreWebForms#59]
Hi Everyone,
We are trying to see if there is a genuine interest for devs or companies to move Webform apps to .NET Core (using ASP.NET Core) as is(without a re-write, with minor modifications) and run across both OS(Windows or Unix). Please give a 👍 reaction to this post. Please don't 👎 anything, this isn't the appropriate place to express negative opinions.

If you wish to have a specific feature to be prioritized, create an issue, add a short comment to this issue with your issue request.

**Comment by AndersMad** [comment]
Wow! This is amazing!! Been searching for this for years and last night I started trying to make base work via ref.source, searched for missing code and I ended up finding this gem! With nuget updates later that Saturday night solving my issues (cheers @twsouthwick !).. So yes, it seems like the project is active 😃 

Q1: Is this official MS work? I see a few MS email people working on it.

Q2: Will it include `System.Web.Extensions`? I tried brutely adding it from ref.source and by the looks of the compile errors - it seems fairly straight forward based on what has been done.. I am mainly missing `UpdatePanel` / AJAX'ing and `IHttpHandlerFactory` (as its used by that too - but not important).

This is the base test of partial update:

```
<%@ Page Async="true" Language="C#" AutoEventWireup="true"  %>

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" >
<head runat="server">
	<title>Ajax Test</title>
	<style>
		* { font-family: Consolas; }
		body { background-color: #222; color: #eee;  }
	</style>
</head>
<body>
<form id="form1" runat="server">

	<script runat="server">
		protected void Button1_Click(object sender, EventArgs e) {
			Label1.Text = "I am postyback!";
		}
	</script>

	<p>Should stay GET after click: <b><% Response.Write(Page.Request.HttpMethod); %> </b></p>

	<asp:ScriptManager EnablePartialRendering="true" ID="ScriptManager1" runat="server"></asp:ScriptManager>
	
	<asp:UpdatePanel ID="UpdatePanel1" runat="server">
		<ContentTemplate>
			<asp:Label ID="Label1" runat="server" Text="Do ->"></asp:Label>
			<asp:Button ID="Button1" runat="server" Text="Click" OnClick="Button1_Click" />
		</ContentTemplate>
	</asp:UpdatePanel>
	
</form>
</body>
</html>
```

At a later stage (I hope there will be), I will try to compile "Telerik UI for ASP.NET AJAX" against this. Possible making the transition to .NET Core / 8 a large task - rather than an impossible one.

**Comment by twsouthwick** [comment]
@AndersMad glad you found it! It's been a pet project of mine for the past year or so. 

> Is this official MS work? I see a few MS email people working on it.

It is not an official work. It's something I'd like to see work, and will make fixes here and there (as you see I've gotten on a kick the last couple of days). I'm the author of dotnet/systemweb-adapters and would love to see what that can enable. However, I'll reiterate that this is a side-project and not an official project (maybe one day like CoreWCF?)

> Will it include `System.Web.Extensions`?

The beginnings of it is already there! Grab `WebForms.Extensions`. I'm putting together an e2e sample with eShop on CoreWebForms/Samples to show how to get it configured. I'm finding a number of issues, so fixing those so the e2e can work :)

> I am mainly missing....

I've used microsoft/referencesource to get the code to fill in WebForms as I need it. There's a lot working now, so additional types can be fairly easily added in. For example, I just hit an issue where I needed ExpressionBuilders and got that hooked up in #115 .

Feel free to open issues for new ideas and/or missing APIs. @birojnayak and I are both interested in seeing this working for people, and as time permits we'll be working on it.

**Comment by AndersMad** [comment]
> The beginnings of it is already there! Grab `WebForms.Extensions`. 

Thanks for the reply! Yes, I saw - but I compared source to ref showing the Ajax stuff missing :) Meanwhile I tried to convert the Telerik solution mentioned to v8 (large control suite with a neat number of 22.222 files actually) and it seems like mostly yielding fairly simple compile errors (2k) from main like missing `DataListItem` and simple enums like `DayNameFormat` etc. And then the things related to the Extensions (`UpdatePanel`, `JavaScriptConverter` in System.Web.Script.Serialization etc.) - but I spose that the Ajax might be a bit hefty port.

> Additional types can be fairly easily added in

As you say and by the looks of it - but I reckon your focused on the complex main. But up-vote on basic from me :)

> Feel free to open issues for new ideas and/or missing

I got that huge thingy to compile against now - so just pling me if you need some testing later on - when your at a more comfy place.

In future - possible more async and less "internal private" to open the black-box for more transitioning. At this point we don't care about micro perf. and "should not".

> It is not an official work

Well I very much hope it will be! This is not about staying on WebForms and I really found it VERY odd that MS did not provide this. In our case it simply impossible to make a binary switch as some customers will stay with certain customizations and version for a very long time. All WebForms related has been nerfed down to a minimum (almost no view state and .aspx etc.) - but a TON of dynamics is going on driven by data (dynamic pages) and a huge admin system - again altered on data. The TCO simply does not allow this when you have a lot of ongoing customers staying for 20+ years. Rant over - and again a big thanks to you and co. 🍻!

**Comment by twsouthwick** [comment]
Can you create issues to track APIs you find you need, such as  `DataListItem`, `DayNameFormat`, etc? It'll help prioritize work. I'm mostly going off what I need to enable some basic samples I have available to sanity check things.

> possible more async and less "internal private" to open the black-box for more transitioning. At this point we don't care about micro perf. and "should not".

can you describe this a bit more?

**Comment by birojnayak** [comment]
Feature :- supporting Ajax/Telerik/Devex controls in System.Web*  (tracked here #9)

**Comment by AndersMad** [comment]
>  I'm mostly going off what I need to enable some basic samples I have available to sanity check things

Yes, I am not assuming your in the need of test cases 😄 I'll make some posts later when I have seen it mature some - for starters I copied my basic sample to #9 

> can you describe this a bit more?

Not at all priority! Just me thinking far ahead on a late Sunday night! But many things regarding async seemed incomplete like events and "overriding" HTML and JS seemed hackish - but tbo it's a long time ago, but tried stripping WebForms usage as much as possible - just remember it was hard at sometimes and I went changing things via reflections.

**Comment by julealgon** [comment]
Are there any plans to actually introduce new features and bug fixes on top of original WebForms? I actually ran into a _nasty_ limitation with DynamicData and would absolutely love to have that eventually fixed.

Let me know if documenting the issue/scenario in this repo would be beneficial.

**Comment by twsouthwick** [comment]
> Are there any plans to actually introduce new features and bug fixes on top of original WebForms? I actually ran into a _nasty_ limitation with DynamicData and would absolutely love to have that eventually fixed.
> 
> Let me know if documenting the issue/scenario in this repo would be beneficial.



> Are there any plans to actually introduce new features and bug fixes on top of original WebForms? I actually ran into a _nasty_ limitation with DynamicData and would absolutely love to have that eventually fixed.
> 
> Let me know if documenting the issue/scenario in this repo would be beneficial.



> Are there any plans to actually introduce new features and bug fixes on top of original WebForms? I actually ran into a _nasty_ limitation with DynamicData and would absolutely love to have that eventually fixed.
> 
> Let me know if documenting the issue/scenario in this repo would be beneficial.

Feel free to create an issue and depending on what the fix is, that probably is not out of scope here

**Comment by gumbarros** [comment]
Checking this repo after a few months.

Is someone using at production already?

**Comment by simonegli8** [comment]
Hi, I have a similar project https://github.com/simonegli8/WebFormsForCore wich is a direct port of System.Web to NET 8, and it allows WebForms projects to directly run on NET 8.

**Comment by birojnayak** [comment]
@simonegli8 really nice that you have also build some form of replacement ... see if you want to port some of the functionalities into this project to help others. Our plan is enrich this one to make a default replacement (for Webform in Linux/Windows) which whole community can benefit from.

**Comment by AndersMad** [comment]
nice @simonegli8 !👏  I haven't checked the basis on how it compares to this prject - but I did take a quick look on some missing features in this project (that I stated here https://github.com/CoreWebForms/CoreWebForms/issues/300#issuecomment-2106218733) and it seems you completed them all 👌
